proposals/0383-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,179 @@
+# Allow Property Wrappers on Let Declarations
+
+* Proposal: [SE-038N](NNNN-filename.md)
+* Authors: [Amritpan Kaur](https://github.com/amritpan), [Pavel Yaskevich](https://github.com/xedin)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: [apple/allow-let-property-wrapper](https://github.com/apple/swift/pull/62342) 
+
+## Introduction
+
+SE-0258 Property Wrappers models both mutable and immutable variants of a property wrapper but does not allow wrapped properties explicitly marked as a `let`. This proposal builds on the original vision by expanding property wrapper usage to include let declared properties with a wrapper attribute.
+
+## Motivation
+
+Allowing property wrappers to be applied to `let` properties improves Swift language consistency and code safety.
+
+Today, a property wrapper can be applied to a property of any type but the rules for declaring these wrapped properties are varied and singular to property wrappers. For example, a struct instance without the property wrapper attribute can be declared with either a `var` or a `let` property. However, mark the struct as a property wrapper type and the compiler no longer allows it to be written as a `let` wrapped property:
+```
+@propertyWrapper
+struct Wrapper {
+  var wrappedValue: Int { 0 }
+  init(wrappedValue: Int) {}
+}
+
+struct S {
+  @Wrapper let value: Int // Error: Property wrapper can only be applied to a ‘var’
+}
+```
+Permitting wrapped properties to mimic the rules for other type instances that can be written with either a `var` or a `let` will simplify the Swift language.
+
+Additionally, `let` wrapped properties add code safety where a user wants to expressly remove access to a property’s mutators after initializing the property once or simply does not need a mutable property wrapper. This could be useful for property wrappers that do not change or are reference types.
```

Comment:
See comment above regarding the use of "safety."

Reply:
Additionally, I would suggest consider incorporating the discussion in the pitch thread prompted by Gustaf's question: "What kinds of interesting behavior would this actually allow that we can't have now?"

The answers in the thread, beyond the point you mention above, are so far:
* Reference semantics if property wrapper implementation supports it (Pavel).
* `nonisolated` wrapped properties (Holly).

Although you mention reference semantics below, I would suggest it's worth calling it out here at the top. You may want to consider if there are other "interesting behaviors" that would fit the list.